v2/tools/generator/internal/codegen/pipeline/add_arm_conversion_interface.go

```diff
@@ -198,8 +198,12 @@ func (c *armConversionApplier) transformSpec(resourceType *astmodel.ResourceType
 			return t, nil
 		}
 
-		// rename Name to AzureName
-		azureNameProp := armconversion.GetAzureNameProperty(c.idFactory).WithType(nameProp.PropertyType())
+		// rename Name to AzureName and promote type if needed
+		namePropType := nameProp.PropertyType()
+		if optional, ok := namePropType.(*astmodel.OptionalType); ok {
+			namePropType = optional.Element()
+		}
```

Comment:
Might this ever be wrapped in another type?

Reply:
At least currently, no.
If it were, we would need to make this modification with a visitor to maintain the wrapping. I've updated the code to leave a note to this effect but left it as is for now.